code/modules/mob/living/carbon/xenomorph/xeno_defines.dm

```diff
@@ -934,6 +934,24 @@
 	else
 		hive_ui.update_burrowed_larva()
 
+/datum/hive_status/proc/respawn_on_turf(client/xeno_client, turf/spawning_turf)
+	var/mob/living/carbon/xenomorph/larva/new_xeno = spawn_hivenumber_larva(spawning_turf, hivenumber)
+	if(isnull(new_xeno))
+		return FALSE
+
+	if(!SSticker.mode.transfer_xeno(xeno_client,.mob, new_xeno))
+		qdel(new_xeno)
+		return FALSE
+
+	new_xeno.visible_message(SPAN_XENODANGER("A larva suddenly emerge from a dead husk!"),
+	SPAN_XENOANNOUNCE("The hive has no core! You manage to emerge from your old husk as a larva!"))
+	msg_admin_niche("[key_name(new_xeno)] burrowed out from \a [spawning_turf]. [ADMIN_JMP(spawning_turf)]")
```

Comment:
```suggestion
	new_xeno.visible_message(SPAN_XENODANGER("A larva suddenly emerges from a dead husk!"),\
	SPAN_XENOANNOUNCE("The hive has no core! You manage to emerge from your old husk as a larva!"))
	msg_admin_niche("[key_name(new_xeno)] respawned at \a [spawning_turf]. [ADMIN_JMP(spawning_turf)]")
```

Reply:
These being the default messages now though makes this not a very generic proc; but until something else uses it... I guess it doesn't matter.